helix-core/src/main/java/org/apache/helix/task/TaskStateModelFactory.java

```diff
@@ -41,23 +47,15 @@
 
   private final HelixManager _manager;
   private final Map<String, TaskFactory> _taskFactoryRegistry;
-  private final ScheduledExecutorService _taskExecutor;
+  private ScheduledExecutorService _taskExecutor;
```

Comment:
Update:

Neal and I had an offline discussion. To simplify the design, we will create a temporary ZK connection to read the task pool size configs and close right away. This would greatly simplify the code and this won't be that much of a burden since the connection is closed right away.

This is to populate the current task size pool at the time of LiveInstance (ephemeral node) initialization.

@NealSun96 Please remember to put a TODO here to potentially revisit the logic - ideally, we shouldn't have to create multiple ZK client connections when there already is a manager there.

Reply:
Fyi @jiajunwang